core/src/main/scala/kafka/message/Message.scala

```diff
@@ -36,83 +39,156 @@ object Message {
   val MagicLength = 1
   val AttributesOffset = MagicOffset + MagicLength
   val AttributesLength = 1
-  val KeySizeOffset = AttributesOffset + AttributesLength
+  // Only message format version 1 has the timestamp field.
+  val TimestampOffset = AttributesOffset + AttributesLength
+  val TimestampLength = 8
+  // Message format version 0 does not have timestamp field.
+  val KeySizeOffset_V0 = AttributesOffset + AttributesLength
+  // Message format version 1 has timestamp field.
+  val KeySizeOffset_V1 = TimestampOffset + TimestampLength
   val KeySizeLength = 4
-  val KeyOffset = KeySizeOffset + KeySizeLength
+  // Message format version 0 does not have timestamp field.
+  val KeyOffset_V0 = KeySizeOffset_V0 + KeySizeLength
+  // Message format version 1 has timestamp field.
```

Comment:
Do we really need to mention the same thing so many times? It seems to me that it would be enough to mention once that message format 0 does not have a timestamp field and message format 1 does.